### PR TITLE
Support PKCS#11 for mutual TLS on Unix platforms

### PR DESCRIPTION
**Pull request by graebm**
Add bindings for new PKCS#11 functionality.

Related: https://github.com/awslabs/aws-c-io/pull/451

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/io.c [hunk 1]
```diff
@@ -342,24 +466,100 @@ napi_value aws_napi_io_tls_ctx_new(napi_env env, napi_callback_info info) {
...
+                    goto cleanup;
+                }
+                pkcs11_options.slot_id = &pkcs11_slot_id;
+            }
+        }
+
+        /* clang-format off */
+
+        #define PARSE_PKCS11_OPTION_STR(property_name, option_cursor, storage_buffer)                                  \
+        do {                                                                                                           \
```
Comment (bretambrose) [inline source/io.c:543]: Can't we just use a scoped block here?

File: source/logger.c [hunk 2]
```diff
@@ -213,29 +207,47 @@ static void s_threadsafe_log_call(napi_env env, napi_value node_log_fn, void *co
...
+        goto done;
+    }
+
     napi_value node_process = NULL;
-    AWS_NAPI_CALL(env, napi_get_named_property(env, node_global, "process", &node_process), { goto done; });
+    if (napi_ok != napi_get_named_property(env, node_global, "process", &node_process)) {
+        goto done;
+    }
+
+    for (struct aws_linked_list_node *list_node = aws_linked_list_rbegin(&msgs);
```
Comment (bretambrose) [inline source/logger.c:223]: I'd add a comment note for why reverse iteration is correct

**Review by bretambrose — APPROVED** [review]

File: source/io.c (hunk 1 above)
Comment (graebm) [reply source/io.c:543]: it's the classic "put your macro in a `do{}while(0)` so users are forced to put semicolon after it" trick

but yeah, confusing, and this macro not a global helper

I'll simplify

File: source/logger.c (hunk 2 above)
Comment (graebm) [reply source/logger.c:223]: 😱 OH NO THIS IS NOT CORRECT AT ALL
great catch. I have failed this interview

**Pull request merged**